### PR TITLE
Smusand2/proxyreaderendpoint

### PR DESCRIPTION
**Pull request by samsanders23**
Re-routing Lambda requests to the RDS Proxy reader endpoint instead, to utilise a new RDS Proxy feature. This also creates 2 Aurora readers to route the read requests to. Tested and is functional.

Also the CloudWatch dashboard was updated to reduce the period interval from 300 to 60 seconds

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cfn/dashboard.template [hunk 1]
```diff
@@ -29,31 +29,31 @@ Resources:
           - !Ref ProxyAccessLambda
           - '", ".", "'
           - !Ref ProxyAccessLambda
-          - !Sub '", {"label": "Proxy"}]], "region": "${AWS::Region}", "title": "Lambda Concurrent executions", "view": "timeSeries", "stacked": false, "period": 300, "stat": "Maximum", "yAxis": {"left": {"min": 0}}}}, {"type": "metric", "x": 0, "y": 6, "width": 6, "height": 6, "properties": {"metrics": [["AWS/Lambda", "Invocations", "FunctionName", "'
+          - !Sub '", {"label": "Proxy"}]], "region": "${AWS::Region}", "title": "Lambda Concurrent executions", "view": "timeSeries", "stacked": false, "period": 60, "stat": "Maximum", "yAxis": {"left": {"min": 0}}}}, {"type": "metric", "x": 0, "y": 6, "width": 6, "height": 6, "properties": {"metrics": [["AWS/Lambda", "Invocations", "FunctionName", "'
```
Comment (rezabekf) [inline cfn/dashboard.template:32]: Are that data send to dashboard every minute? Thought the default time is every 5 minutes.

File: cfn/rds.template [hunk 2]
```diff
@@ -78,25 +68,35 @@ Resources:
...
+
+  DBInstance3:
+    Type: AWS::RDS::DBInstance
+    Properties:
+      DBInstanceClass: !Ref DBInstanceClass
+      EnablePerformanceInsights: true
+      PerformanceInsightsRetentionPeriod: !Ref PerformanceInsightsRetentionPeriod
+      DBClusterIdentifier: !Ref DBCluster
+      Engine: aurora-mysql
+      PubliclyAccessible: false
```
Comment (rezabekf) [inline cfn/rds.template]: What are these additional DBs used for? Are they read instances? If yes, can you Change naming to, for example `DbReadInstance1` | `DbWriteInstance1`

File: src/functions/noproxy/cr_add_users_noproxy.py [hunk 3]
```diff
@@ -27,7 +27,6 @@
 
 @helper.create
 def create(event, context):
-    logger.info("Got Create")
```
Comment (rezabekf) [inline src/functions/noproxy/cr_add_users_noproxy.py]: why is this deleted?

File: src/functions/proxy/cr_add_users_proxy.py [hunk 4]
```diff
@@ -40,7 +40,6 @@
 
 @helper.create
 def create(event, context):
-    logger.info("Got Create")
```
Comment (rezabekf) [inline src/functions/proxy/cr_add_users_proxy.py]: why was this removed?

File: cfn/rds.template [hunk 5]
```diff
@@ -184,29 +196,14 @@ Resources:
...
-                    - - !Sub arn:${AWS::Partition}:kms:${AWS::Region}:${AWS::AccountId}:key/
-                      - !Select
-                        - 6
-                        - !Split
-                          - ':' # Get the DB Secret name without arn bit
-                          - !Ref DBSecret
-                  - !Sub arn:${AWS::Partition}:kms:${AWS::Region}:${AWS::AccountId}:key/*
-                Condition:
-                  StringEquals:
-                    kms:ViaService: !Sub secretsmanager.${AWS::Region}.amazonaws.com
```
Comment (rezabekf) [inline cfn/rds.template]: wasn't this removed in PR #16 ?

**Review by rezabekf — CHANGES_REQUESTED** [review]

File: cfn/dashboard.template (hunk 1 above)
Comment (samsanders23) [reply cfn/dashboard.template:32]: The granularity can be increased to every minute.

File: cfn/rds.template (hunk 2 above)
Comment (samsanders23) [reply cfn/rds.template]: Yes, I'll update this accordingly

File: src/functions/noproxy/cr_add_users_noproxy.py (hunk 3 above)
Comment (samsanders23) [reply src/functions/noproxy/cr_add_users_noproxy.py]: Removed as it's unnecessary when the lambda logs are analysed, the message 'got create' is confusing. This won't be needed in the final form

**Comment by samsanders23** [comment]
Also updated CW Dashboard accordingly, to use the reader endpoints, rather than the aurora writer instance.

File: src/functions/noproxy/cr_add_users_noproxy.py (hunk 3 above)
Comment (rezabekf) [reply src/functions/noproxy/cr_add_users_noproxy.py]: Its actual opposite, the logging is there to help to see what function within lambda was called, e.g. on creating stack you will see message "Got Create", on updating stack "Got Update and on deleting stack "Got Delete", I dont see the reason to remove this, as it helps troubleshoot `crhelper` calls in CWL. For more info see https://pypi.org/project/crhelper/

File: src/functions/proxy/cr_add_users_proxy.py (hunk 4 above)
Comment (rezabekf) [reply src/functions/proxy/cr_add_users_proxy.py]: Same as above

File: .pre-commit-config.yaml [hunk 6]
```diff
@@ -42,11 +42,11 @@ repos:
...
-# - repo: https://github.com/aws-cloudformation/rain
-#   rev: v1.2.0
-#   hooks:
-#   - id: cfn-format
-#     files: \.template$
+- repo: https://github.com/aws-cloudformation/rain
+  rev: v1.2.0
+  hooks:
+  - id: cfn-format
+    files: \.template$
```
Comment (rezabekf) [inline .pre-commit-config.yaml:49]: Perfect

**Review by rezabekf — CHANGES_REQUESTED** [review]

File: src/functions/noproxy/cr_add_users_noproxy.py (hunk 3 above)
Comment (samsanders23) [reply src/functions/noproxy/cr_add_users_noproxy.py]: I suppose I'm confused by the wording, what do you mean by 'Got create'? Do you mean 'Create function initialized', or 'Users created'? If so I'll change to this (and the delete and update statements accordingly) and update the PR

Comment (rezabekf) [reply src/functions/noproxy/cr_add_users_noproxy.py]: It's relevant to the function initialized. Check the link I have attached.

Comment (samsanders23) [reply src/functions/noproxy/cr_add_users_noproxy.py]: I see the purpose, just not the wording. Added back anyway.

**Review by rezabekf — APPROVED** [review]
LGMT

**Pull request merged**